sphinx/domains/__init__.py

```diff
@@ -212,7 +212,7 @@ def role(self, name):
         fullname = '%s:%s' % (self.name, name)
 
         def role_adapter(typ, rawtext, text, lineno, inliner, options={}, content=[]):
-            # type: (unicode, unicode, unicode, int, Inliner, Dict, List[unicode]) -> nodes.Node  # NOQA
+            # type: (unicode, unicode, unicode, int, Inliner, Optional[Dict], Optional[List[unicode]]) -> Tuple[List[nodes.Node], List[nodes.Node]]  # NOQA
```

Comment:
Sorry for my too-late-comment after merging, but added two `Optional` seems unnecessary.

Reply:
Thanks, I'd confused.